src/scanner.rs

```diff
@@ -78,29 +82,48 @@ impl Scanner {
-
-                const ETH_GET_BLOCK_NUMBER_METHOD: &str = "eth_blockNumber";
-                const ETH_GET_LOGS_METHOD: &str = "eth_getLogs";
-
-                let mut batched_requests_builder = BatchRequestBuilder::new();
-
-                batched_requests_builder
-                    .insert(ETH_GET_BLOCK_NUMBER_METHOD, Vec::<String>::new()).map_err(|err| ScannerError::AddCallToBatch {
-                        method: ETH_GET_BLOCK_NUMBER_METHOD.to_owned(),
-                        source: err,
-                    })?;
-
-                batched_requests_builder
-                    .insert(ETH_GET_LOGS_METHOD, vec![filter]).map_err(|err| ScannerError::AddCallToBatch {
-                        method: ETH_GET_LOGS_METHOD.to_owned(),
-                        source: err,
-                    })?;
-
-                let responses = self.client
+                let perform = || async {
+                    let filter = self.filter.clone().from_block(self.from_block_number).to_block(BlockNumber::Latest);
+                    tracing::debug!("fetching new logs from block {:?} to {:?}", filter.get_from_block(), filter.get_to_block());
+
+                    let mut batched_requests_builder = BatchRequestBuilder::new();
+
+                    batched_requests_builder
+                        .insert(ETH_GET_BLOCK_NUMBER_METHOD, Vec::<String>::new()).map_err(|err| backoff::Error::Transient {
+                            err: ScannerError::AddCallToBatch {
+                                method: ETH_GET_BLOCK_NUMBER_METHOD.to_owned(),
+                                source: err,
+                            },
+                            retry_after: None,
+                        })?;
+
+                    batched_requests_builder
+                        .insert(ETH_GET_LOGS_METHOD, vec![filter]).map_err(|err| backoff::Error::Transient {
+                            err: ScannerError::AddCallToBatch {
+                                method: ETH_GET_LOGS_METHOD.to_owned(),
+                                source: err,
+                            },
+                            retry_after: None,
+                        })?;
+
+                    self.client
                     .batch_request::<Value>(batched_requests_builder)
-                    .await.map_err(|err| ScannerError::FilterUpdate(err))?;
+                    .await.map_err(|err| {backoff::Error::Transient {
+                        err: ScannerError::FilterUpdate(err),
+                        retry_after: None,
+                    }})
+                };
+
+                let responses = backoff::future::retry(
+                    ExponentialBackoffBuilder::new()
+                        .with_max_elapsed_time(Some(
+                            self.interval.period() / 2,
+                        ))
+                        .build(),
+                        perform,
+                )
+                .await?;
```

Comment:
Why we don't log in case of error here?

Reply:
There's not logging in general in the scanner. Errors are thrown and logged by the caller here (the stream consumer).